pyracf/resource/resource_admin.py

```diff
@@ -285,6 +285,134 @@ def get_user_access(
+        if "failures" in audit_rules:
+            traits[f"base:audit_{audit_rules['failures']}"] = "FAILURE"
+        if "all" in audit_rules:
+            traits[f"base:audit_{audit_rules['all']}"] = "ALL"
+        if success is not None:
+            traits[f"base:audit_{success}"] = "SUCCESS"
+        if failure is not None:
+            traits[f"base:audit_{failure}"] = "FAILURE"
+        if all is not None:
+            traits[f"base:audit_{all}"] = "ALL"
+        result = self.alter(resource, class_name, traits=traits)
+        return self._to_steps(result)
+
+    def overwrite_audit_by_access_level(
+        self,
+        resource: str,
+        class_name: str,
+        alter: Union[str, None] = None,
+        control: Union[str, None] = None,
+        read: Union[str, None] = None,
+        update: Union[str, None] = None,
+    ) -> Union[dict, bytes]:
+        """
+        Overwrites the auditing rules for this general resource profile with new
+        rules to audit based on specified access levels.
+        """
+        traits = {}
+        if alter is not None:
+            traits["base:audit_alter"] = alter
+        if control is not None:
+            traits["base:audit_control"] = control
+        if read is not None:
+            traits["base:audit_read"] = read
+        if update is not None:
+            traits["base:audit_update"] = update
+        result = self.alter(resource, class_name, traits=traits)
+        return self._to_steps(result)
+
+    def alter_audit_by_access_level(
+        self,
+        resource: str,
+        class_name: str,
+        alter: Union[str, None] = None,
+        control: Union[str, None] = None,
+        read: Union[str, None] = None,
+        update: Union[str, None] = None,
+    ) -> Union[dict, bytes]:
+        """
+        Alters the auditing rules for this general resource profile with a new
+        rule to audit alter access, preserving existing non-conflicting rules.
+        """
+        result = [self.extract(resource, class_name)]
+        profile = result[0]["securityResult"]["resource"]["commands"][0]["profiles"][0]
+        audit_rules = self._get_field(profile, "base", "auditing")
+        traits = {}
+        if "success" in audit_rules:
+            traits[f"base:audit_{audit_rules['success']}"] = "SUCCESS"
+        if "failures" in audit_rules:
+            traits[f"base:audit_{audit_rules['failures']}"] = "FAILURE"
+        if "all" in audit_rules:
+            traits[f"base:audit_{audit_rules['all']}"] = "ALL"
+        if alter is not None:
+            traits["base:audit_alter"] = alter
+        if control is not None:
+            traits["base:audit_control"] = control
+        if read is not None:
+            traits["base:audit_read"] = read
+        if update is not None:
+            traits["base:audit_update"] = update
+        result = self.alter(resource, class_name, traits=traits)
+        return self._to_steps(result)
+
+    def clear_all_audit_rules(
```

Comment:
I think it might make sense to rename `clear_all_audit_rules` to `remove_all_audit_rules` because it is more consistent with other function names like `remove_all_class_authorizations`.

Reply:
agree this is better. change in next commit.